### ISSUE TITLE
[Impeller] Run dart:gpu tests on CI.

### ISSUE THREAD

**Issue by chinmaygarde** [flutter/flutter#127712]
This issue tracks running the Impeller Scene and dart:gpu tests on CI as we continue to make progress on those prototypes. Per @zanderso, the Engine V2 build setup should make configuring such a shard somewhat straightforward.

**Comment by chinmaygarde** [comment]
cc @bdero

**Comment by bdero** [comment]
Landed a patch that tests the API on CI -- however, it's not possible to run Impeller through flutter_tester right now. So it needs to be rewritten as either a GL or Vulkan embedder.

**Comment by chinmaygarde** [comment]
Can we stash working on the tester and focus on a playgrounds like environment?

**Comment by bdero** [comment]
Yeah, I think that's a great idea actually. I'll see if I can pick up on that Dart Impeller playground setup we added some time ago, and proceed to just use `flutter_tester` to test that the right exceptions get thrown when _not_ using Impeller.

**Comment by bdero** [comment]
Looked into this a bit, and I think for testing in the playground, I'm going to opt for simply backdooring the API to hand it the playground's Impeller context (as opposed to spinning up the whole engine and using the normal flow of fetching the context from the IO manager, etc).
